### PR TITLE
Instanced Rendering first pass

### PR DESCRIPTION
**Pull request by wejdell**
Camera tweaks

### CODE REVIEW

File: Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp [hunk 1]
```diff
@@ -56,42 +58,47 @@ namespace Havtorn
...
-				components[static_cast<U8>(EComponentType::StaticMeshComponent)] = staticMeshComponent;
-				components[static_cast<U8>(EComponentType::DirectionalLightComponent)] = directionalLightComponents[0];
-				SRenderCommand command(components, ERenderCommandType::ShadowAtlasPrePassDirectional);
-				RenderManager->PushRenderCommand(command);
-			}
+			if (!RenderManager->IsStaticMeshInInstancedRenderList(staticMeshComponent->Name)) // if static, if instanced
+			{		
+				if (!directionalLightComponents.empty())
+				{
+					std::array<Ref<SComponent>, static_cast<size_t>(EComponentType::Count)> components;
```
Comment (Aki-Gonzalez) [inline Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp:65]: Is it not possible to turn this segment into a function, there are onlya few differences between directional, point and spot.

Comment (Aki-Gonzalez) [reply Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp:65]: Referring to ex: lines 65-70 == 75-80 == 85-90

Comment (wejdell) [reply Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp:65]: It is, wanted to do a first pass before checking whether function calls will be too detrimental/slow or not

Comment (wejdell) [reply Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp:65]: There is a bug hiding as well so was gonna look at it next

File: Havtorn/Source/Engine/Graphics/RenderManager.h [hunk 2]
```diff
@@ -167,6 +174,18 @@ namespace Havtorn
 			Context->Unmap(buffer, 0);
 		}
 
+		template<class T>
+		void BindBuffer(ID3D11Buffer* buffer, const std::vector<T>& bufferData, std::string bufferType)
```
Comment (Aki-Gonzalez) [inline Havtorn/Source/Engine/Graphics/RenderManager.h]: 178: std::string could be a reference?

File: Havtorn/Source/Engine/Graphics/Shaders/DeferredAnimation_VS.hlsl [hunk 3]
```diff
@@ -38,9 +38,9 @@ VertexModelToPixel main(VertexModelInput input)
     input.Position.x = skinnedPos.x;
     input.Position.y = skinnedPos.y;
     input.Position.z = skinnedPos.z;
-    input.Position.w = skinnedPos.w;
+    //input.Position.w = skinnedPos.w;
```
Comment (Aki-Gonzalez) [inline Havtorn/Source/Engine/Graphics/Shaders/DeferredAnimation_VS.hlsl]: Please remove or add explanation to why L41 is commented.

File: Havtorn/Source/Engine/Graphics/Shaders/Includes/DeferredShaderStructs.hlsli [hunk 4]
```diff
@@ -58,7 +79,7 @@ struct VertexPaintModelToPixel
     float4 WorldPosition    : WORLDPOSITION;
     float4 Normal           : NORMAL;
     float4 Tangent          : TANGENT;
-    float4 Binormal         : BINORMAL;
+    float4 Binormal : BINORMAL;
```
Comment (Aki-Gonzalez) [inline Havtorn/Source/Engine/Graphics/Shaders/Includes/DeferredShaderStructs.hlsli:82]: (Optional) Please fix the indentation.

**Review by Aki-Gonzalez — APPROVED** [review]

**Pull request merged**